File: algo/problems/0242-valid-anagram/go/validanagram_test.go
package validanagram

import "testing"

func TestIsAnagram(t *testing.T) {
	cases := []struct {
		s, t string
		want bool
	}{
		{"anagram", "nagaram", true},
		{"rat", "car", false},
		{"a", "ab", false},
	}
	for _, c := range cases {
		if got := IsAnagram(c.s, c.t); got != c.want {
			t.Errorf("IsAnagram(%q, %q) = %v, want %v", c.s, c.t, got, c.want)
		}
	}
}

File: algo/problems/0242-valid-anagram/java/ValidAnagramTest.java
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

public class ValidAnagramTest {
    @Test
    void valid() { assertTrue(ValidAnagram.isAnagram("anagram", "nagaram")); }

    @Test
    void invalid() { assertFalse(ValidAnagram.isAnagram("rat", "car")); }

    @Test
    void diffLen() { assertFalse(ValidAnagram.isAnagram("a", "ab")); }
}
